tests/test_scdataframe.py

```diff
@@ -0,0 +1,101 @@
+"""
+Tests cosmicqc SCDataFrame module
+"""
+
+import pathlib
+
+import pandas as pd
+from cosmicqc.scdataframe import SCDataFrame
+from pyarrow import parquet
+
+
+def test_SCDataFrame_with_dataframe(
+    tmp_path: pathlib.Path,
+    basic_outlier_dataframe: pd.DataFrame,
+    basic_outlier_csv: str,
+    basic_outlier_csv_gz: str,
+    basic_outlier_tsv: str,
+    basic_outlier_parquet: str,
+):
+    # Tests SCDataFrame with pd.DataFrame input.
+    sc_df = SCDataFrame(data=basic_outlier_dataframe)
+
+    # test that we ingested the data properly
+    assert sc_df.data_source == "pandas.DataFrame"
+    assert sc_df.equals(basic_outlier_dataframe)
```

Comment:
Since you can also output not just a pd.DataFrame, do we need to include tests to make sure the class outputs the other types correctly too (e.g., representation, etc)?

Reply:
I guess my confusion is why the Class has the option to output not just a pd.DataFrame, if all we need for coSMicQC is the pd.DataFrame?

Reply:
Thanks for this comment, I think you bring up a good point. Mostly I was concerned about tracking data provenance and adding new capabilities which might not be entirely reliant on a pd.DataFrame, but could benefit from internal use within the class through a pd.DataFrame. Another way of thinking about this: all we need in `coSMicQC right now is the `pd.DataFrame`, but we will need other output later (as in #2 and #33). In my opinion, this builds a path towards those things using a similar data structure.

Generally the string-based representation is tested through the CLI (where we have to use a string instead of the object in a terminal) and to this effect we end up with tests for this in #14 (where we get a chance to test the string-based repr as well as the CLI).